File: app/src/main/java/com/niiadotei/storehouse/ui/customers/purchases/CustomerPurchasesActivity.java
package com.niiadotei.storehouse.ui.customers.purchases;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.niiadotei.storehouse.data.DatabaseHelper;
import com.niiadotei.storehouse.databinding.ActivityCustomerPurchasesBinding;

public class CustomerPurchasesActivity extends AppCompatActivity {
    RecyclerView recyclerView;

    CustomerPurchasesViewModel customerPurchasesViewModel;

    DatabaseHelper databaseHelper;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ActivityCustomerPurchasesBinding binding = ActivityCustomerPurchasesBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setTitle("Customer Purchases");

        ActionBar actionBar = getSupportActionBar();
        assert actionBar != null;
        actionBar.setDisplayHomeAsUpEnabled(true);

        recyclerView = binding.customerPurchasesRecyclerView;

        databaseHelper = new DatabaseHelper(getApplicationContext());

        Intent intent = getIntent();
        String customer = intent.getStringExtra("customer");

        recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
        customerPurchasesViewModel = new CustomerPurchasesViewModel(this, databaseHelper.getPurchasesArray(customer));

        recyclerView.setAdapter(customerPurchasesViewModel);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            this.finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/niiadotei/storehouse/ui/customers/purchases/CustomerPurchasesActivity.kt
package com.niiadotei.storehouse.ui.customers.purchases

import androidx.appcompat.app.AppCompatActivity
import com.niiadotei.storehouse.data.DatabaseHelper
import android.os.Bundle
import android.view.MenuItem
import androidx.recyclerview.widget.LinearLayoutManager
import com.niiadotei.storehouse.databinding.ActivityCustomerPurchasesBinding

class CustomerPurchasesActivity : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivityCustomerPurchasesBinding.inflate(layoutInflater)

        setContentView(binding.root)

        title = "Customer Purchases"

        val actionBar = supportActionBar!!
        actionBar.setDisplayHomeAsUpEnabled(true)

        val recyclerView = binding.customerPurchasesRecyclerView
        val databaseHelper = DatabaseHelper(applicationContext)

        val intent = intent
        val customer = intent.getStringExtra("customer")

        recyclerView.layoutManager = LinearLayoutManager(applicationContext)
        val customerPurchasesViewModel = CustomerPurchasesViewModel(databaseHelper.getPurchasesArray(customer!!))
        recyclerView.adapter = customerPurchasesViewModel
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }

        return super.onOptionsItemSelected(item)
    }
}